src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Baidu
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="BaiduAuthenticationHandler"/>.
+    /// </summary>
+    public class BaiduAuthenticationOptions : OAuthOptions
+    {
+        public BaiduAuthenticationOptions()
+        {
+            ClaimsIssuer = BaiduAuthenticationDefaults.Issuer;
+            CallbackPath = new PathString(BaiduAuthenticationDefaults.CallbackPath);
+
+            AuthorizationEndpoint = BaiduAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = BaiduAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = BaiduAuthenticationDefaults.UserInformationEndpoint;
+            
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "uid");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "uname");
+            ClaimActions.MapCustomJson(BaiduAuthenticationConstants.Claims.Portrait,
+                user => $"https://tb.himg.baidu.com/sys/portrait/item/{user.Value<string>("portrait")}");
```

Comment:
The value should be URI-encoded before being formatted into the URI.

Reply:
by the official document, the value for `portrait` is a url-friendly string.

Reply:
Even if it is, it would be safer to URI encode it anyway, then if that changes in the future we won't have to fix anything.